lib/repository.rb

```diff
@@ -0,0 +1,101 @@
+require 'http'
+require 'json'
+
+class Repository
+  def initialize(repo_url, username = nil, password = nil)
+    @username = username || ENV['USERNAME']
+    @password = password || ENV['PASSWORD']
+    @repo_url = repo_url
+    if !match(@repo_url) || @match['hostname'] != 'github.com'
+      puts 'Your Repository URL is invalid.'
+      exit
+    end
+  end
+
+  def get_object(endpoint)
+    if @username && @password
+      object = JSON.parse(HTTP.basic_auth(user: @username, pass: @password).get(endpoint).to_s)
+    else
+      object = JSON.parse(HTTP.get(endpoint).to_s)
+    end
+    object
+  end
+
+  def get_all_objects(endpoint)
+    objects = []
+    page = 1
+    while (partial_objects = get_object("#{endpoint}?page=#{page}&per_page=100")) &&
+      partial_objects.length > 0
+      objects.concat(partial_objects)
+      page += 1
+    end
+    objects
+  end
+
+  def match(repo_url)
+    @match = repo_url.match(/\A(?<scheme>http|https):\/\/(?<hostname>\w+\.\w+)\/(?<username>\w+)\/(?<repo>[\w\d\-]+)/)
+    @match
+  end
```

Comment:
Nice regex!
There is no need to assign this to a variable to return it in the next line though.

```suggestion
  def match(repo_url)
    repo_url.match(/\A(?<scheme>http|https):\/\/(?<hostname>\w+\.\w+)\/(?<username>\w+)\/(?<repo>[\w\d\-]+)/)
  end
```